### PR TITLE
Refactor: Move Admin UI repository code into Controllers

### PR DESCRIPTION
**Pull request by alaca**
Resolves #

## Description

This PR refactors the Donors, Donations, and DonationForms repositories and it moves the _request logic_ of each repository into the separate controller class. Also, the code is updated to use some of the PHP 7.0 features.

## Affects

 Donors, Donations, and DonationForms repositories

## Testing Instructions

These changes only affect the list tables of Donors, Donations, and Donation Forms pages, so to test this, it should be enough to simply visit these pages and check that everything is working correctly ( list table data is loaded, filters are working, etc. )

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/DonationForms/Repositories/DonationFormsRepository.php [hunk 1]
```diff
@@ -3,119 +3,25 @@
...
-                       ->selectRaw('SELECT COUNT(ID) as count')
-                       ->leftJoin('give_donationmeta', 'ID', 'donation_id')
-                       ->where('meta_key', '_give_payment_form_id')
-                       ->where('meta_value', $formId)
-                       ->get();
+            ->selectRaw('SELECT COUNT(ID) as count')
+            ->leftJoin('give_donationmeta', 'ID', 'donation_id')
+            ->where('meta_key', '_give_payment_form_id')
+            ->where('meta_value', $formId)
+            ->get();
```
Comment (JasonTheAdams) [inline src/DonationForms/Repositories/DonationFormsRepository.php:24]: Can't this get replaced with `count()`?

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Left a code comment. When I tested this I'm immediately getting fatal errors on any of the list screens. Probably shouldn't do that. 😅

File: src/DonationForms/Repositories/DonationFormsRepository.php (hunk 1 above)
Comment (alaca) [reply src/DonationForms/Repositories/DonationFormsRepository.php:24]: > Fatal error: Uncaught TypeError: Give\DonationForms\Endpoints\ListForms::formatAmount(): Argument #1 ($amount) must be of type string, null given, called in /Users/jasonadams/Local Sites/givewp/app/public/wp-content/plugins/give/src/DonationForms/Endpoints/ListForms.php on line 162 and defined in /Users/jasonadams/Local Sites/givewp/app/public/wp-content/plugins/give/src/DonationForms/Endpoints/ListForms.php on line 170

This is great, PHP 5.6 will allow that.

Comment (alaca) [reply src/DonationForms/Repositories/DonationFormsRepository.php:24]: Resolved: 92efa996

**Comment by alaca** [comment]
@JasonTheAdams I refactored this even more. I added `DonationFormsResponseData` DTO and `DonationFormMetaKeys` class so that we are in line with how we do things in Donations, Donors, etc.

**Comment by JasonTheAdams** [comment]
@alaca I just merged in `develop` — where the tests are passing — but they're failing here. Mind looking into this?

**Comment by alaca** [comment]
@JasonTheAdams all checks have passed.

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
@alaca Tests are passing and the code looks good, but I did find that permantently deleting forms isn't working:

Screen Recording 2022-05-04 at 04 46 05 PM

**Comment by alaca** [comment]
@JasonTheAdams resolved 8f16fb1

**Review by JasonTheAdams — APPROVED** [review]
Great work, @alaca!

**Pull request merged**